### ISSUE TITLE
[Request] Make a 0.9 release compatible with modern scikit-learn

### ISSUE THREAD

**Issue by QuentinSoubeyran** [scikit-optimize/scikit-optimize#1071]
Making a 0.9 release, compatible with `scikit-learn>=0.24` and with the new features would be nice.

Here are a list of PR I think would be useful for a `0.9` release:

__`0.9rc1` release__:
- [ ] Merge #971 
- [ ] Merge #1012 
- [ ] Merge #1050 
- [ ] Merge #1055 
- [ ] Merge #1062 
- [ ] Merge #1063 
- [ ] Merge #1066 
- [ ] Merge #1067 
- [ ] Merge #1069 

__0.9.0 release__
- [ ] release and test `0.9rc1`

I have made a test branch on my fork, [`mock0.9rc1`](https://github.com/QuentinSoubeyran/scikit-optimize/tree/mock0.9rc1), for testing all those features. If someone wishes to test all those features & fixes, one can install from source using
```bash
pip install git+https://github.com/QuentinSoubeyran/scikit-optimize.git@mock0.9rc1
```

See also:
#1059

**Comment by rigtorp** [comment]
I would be happy if only current master with the fix to support sklearn 1.0  #1063 would be released.

**Comment by xmatthias** [comment]
I think there should be a interim 0.8.2 release with current master.

From a projects perspective, it's pretty risky to merge a ton of PR's right before a Release - people testing specific PR's or branches (like requested above) will often not have the edge-cases other users encounter.
So i fear that this approach will delay a release for weeks/months, or worse, result in an unstable 0.9.0 release, as some obtruse compatibility-issues between the PR's might appear.

While current master would be pretty stable as far as i can see (the above PR's don't seem to fix anything that wasn't present in 0.8.1 as well).

**Comment by QuentinSoubeyran** [comment]
@kernc with the release of 0.9.0, feel free to close this. I can also update it to serve as information for people coming on the repo. Though an official, pinned issue would probably be better for that.

**Comment by QuentinSoubeyran** [comment]
@RNarayan73 you should use the above `unofficial` branch instead of the old mock branch if you want to test un-merged features.

**Comment by RNarayan73** [comment]
I tested the multimetric feature in the unofficial branch and it works fine.
Thanks, 
Narayan

**Comment by jmnunezd** [comment]
Thanks! the multi-metric is working for me as well on this branch! hope this can be merge in the future to the main branch

**Comment by QuentinSoubeyran** [comment]
I'm afraid the library hasn't been updated for other a year, so this is unlikely. The last 0.9.0 release is quite dated, and there hasn't been any activity from the maintainers since.
